code/modules/food_and_drinks/kitchen_machinery/kitchen_machine.dm

```diff
@@ -289,24 +273,69 @@
 			fail()
 			return
 	else
-		var/halftime = round(recipe.time/10/2)
-		if(!wzhzhzh(halftime))
+		if(!wzhzhzh(5))
 			abort()
 			return
-		if(!wzhzhzh(halftime))
+		if(!wzhzhzh(5))
 			abort()
 			fail()
 			return
-		cooked = recipe.make_food(src)
-		byproduct = recipe.get_byproduct()
-		stop()
-		if(cooked)
-			cooked.forceMove(loc)
-		for(var/i=1,i<efficiency,i++)
-			cooked = new cooked.type(loc)
-		if(byproduct)
-			new byproduct(loc)
+		make_recipes(recipes_to_make)
+
+/obj/machinery/kitchen_machine/proc/choose_recipes()
```

Comment:
These proc names are vague - you should describe what each proc does with a set of comments for each.

Reply:
While I disagree that these are vague (choose recipes clearly chooses recipes, and make recipes makes the recipe it got passed as an argument), I'll add some comments anyways for the less attentive.